packages/teleport-project-generator-react/src/project-template.ts

```diff
@@ -58,5 +60,171 @@ export default {
 };`,
     },
   ],
-  subFolders: [],
+  subFolders: [
+    {
+      name: 'src',
+      files: [
+        {
+          name: 'cms-item',
+          fileType: FileType.JS,
+          content: `import { useEffect, useState, useRef } from "react";
+/**
+* - @param loading - JSX to display when the componet is in loading state.
+* - @param error - JSX to display when the compnent hit a error state
+* - @param children - Use for the renderProps patter to call the parent
+* - @param fetcher - A promise from which we need to derive another three states, loading, error, data
+* - @param params - Params that need to be passed while fetching the resource.
+* - @param initialData - This is used to avoid the loading state, and fetch the new data when there is a change in the params
+*/
+
+const NO_INITIAL_DATA = Symbol(null);
+
+export const CMSListItem = ({
```

Comment:
I notice that these 2 components are the same both in react and next project templates...is there a reason why we did not extract them into some constants?

Reply:
Yes, we can do that. But actually inside react for this PR is just a experiment 👍

Reply:
I dropped total react for now 👍